### PR TITLE
Add hosted documentation

### LINKED ISSUE #267: Add hosted documentation pages
Currently docstrings are auto-generated, but this documentation is not available anywhere but in the source code.

To make docs more easily available, can we add hosted docs e.g. via Documenter.jl that are built by CI?

Related to #168

### PR DESCRIPTION
**Pull request by nickrobinson251**
- I wanted to know what high-level `S3` functions existed and reading the source was too painful (partly due to #279), so this is an attempt to auto-generate docs for the high-level `@service` modules
  - this PR only adds hosted docs, it doesn't attempt to address the existing issues relating to docstrings (#168, #279)
- i've tried to add a docs build step to the CI GithubAction, and i guess we'd need to add a `GITHUB_TOKEN` + `DOCUMENTER_KEY`, but i'm no expert here
- I'd recommend cloning this branch and building the docs locally as part of reviewing. 
  - The index page is basically a copy of the first half of the README
  - The `AWS` page documents exported functions
  - The pages for all `@service` modules are generated; see the code added to `docs/make.jl`
- closes #267

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[deps]
+AWS = "fbe9abb3-538b-5e4e-ba9e-bc94f4f92ebc"
```
Comment (nickrobinson251) [inline docs/Project.toml:2]: Need to add Documenter

File: README.md [hunk 2]
```diff
@@ -1,9 +1,11 @@
 ## AWS.jl
 
-![CI](https://github.com/JuliaCloud/AWS.jl/workflows/CI/badge.svg)
+[![CI](https://github.com/JuliaCloud/AWS.jl/workflows/CI/badge.svg)](https://github.com/JuliaCloud/AWS.jl/actions?query=workflow%3ACI)
 [![Code Style: Blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
 [![ColPrac: Contributor's Guide on Collaborative Practices for Community Packages](https://img.shields.io/badge/ColPrac-Contributor's%20Guide-blueviolet)](https://github.com/SciML/ColPrac)
 
+[![Docs: stable](https://img.shields.io/badge/docs-stable-blue.svg)](https://invenia.github.io/PkgTemplates.jl/stable)
+[![Docs: dev](https://img.shields.io/badge/docs-dev-blue.svg)](https://invenia.github.io/PkgTemplates.jl/dev)
```
Comment (nickrobinson251) [inline README.md:8]: Need to fix these links to point to aws.jl

**Comment by nickrobinson251** [comment]
Documenter.jl's `HTMLWriter.fixlinks!` function misinterprets some regex syntax in docstrings (e.g. [`\"^[A-Za-z_]([A-Za-z0-9]*|[A-Za-z0-9][A-Za-z0-9_]*)\"`](https://github.com/JuliaCloud/AWS.jl/blob/9cd64f42731e50e4a57404dd3f7cc3cd21db5a8b/src/services/iotanalytics.jl#L14)) as markdown links (due to the `[...](...)` syntax), and then posts warnings about them not being valid links:
```sh
[ Info: RenderDocument: rendering document.
[ Info: HTMLWriter: rendering HTML pages.
┌ Warning: invalid local link: unresolved path in services/serverlessapplicationrepository.md
│   link.text =
│    1-element Array{Any,1}:
│     "a-z0-9"
│   link.url = "([a-z0-9]|-(?!-))*[a-z0-9]"
└ @ Documenter.Writers.HTMLWriter ~/.julia/packages/Documenter/lul8Y/src/Writers/HTMLWriter.jl:1823
┌ Warning: invalid local link: unresolved path in services/serverlessapplicationrepository.md
│   link.text =
│    1-element Array{Any,1}:
│     "a-z0-9"
│   link.url = "([a-z0-9]|-(?!-))*[a-z0-9]"
└ @ Documenter.Writers.HTMLWriter ~/.julia/packages/Documenter/lul8Y/src/Writers/HTMLWriter.jl:1823

┌ Warning: invalid local link: unresolved path in services/iotanalytics.md
│   link.text =
│    1-element Array{Any,1}:
│     "A-Za-z_"
│   link.url = "[A-Za-z0-9]*|[A-Za-z0-9][A-Za-z0-9_]*"
└ @ Documenter.Writers.HTMLWriter ~/.julia/packages/Documenter/lul8Y/src/Writers/HTMLWriter.jl:1823
```
I think this is unavoidable and also unproblematic (e.g. these warnings don't cause errors, even with `strict=true`).
So i think we can just ignore these :)

File: docs/make.jl [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    return services_pages
+end
+
+makedocs(;
+    modules=[AWS],
+    repo="https://github.com/JuliaCloud/AWS.jl/blob/{commit}{path}#{line}",
+    sitename="AWS.jl",
+    format=Documenter.HTML(;
+        prettyurls=false,
+        canonical="https://invenia.github.io/AWS.jl",
```
Comment (nickrobinson251) [inline docs/make.jl:66]:
```suggestion
        canonical="https://juliacloud.github.io/AWS.jl",
```

**Comment by nickrobinson251** [comment]
okay, i think this is ready for review :)

**Review by mattBrzezinski — APPROVED** [review]

**Pull request merged**